reactivesocket-transport-tcp/src/main/java/io/reactivesocket/transport/tcp/client/TcpReactiveSocketConnector.java

```diff
@@ -1,81 +1,50 @@
-/**
- * Copyright 2016 Netflix, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
 package io.reactivesocket.transport.tcp.client;
 
-import io.netty.channel.EventLoopGroup;
-import io.reactivesocket.*;
-import io.reactivesocket.internal.rx.EmptySubscription;
-import io.reactivesocket.rx.Completable;
+import io.netty.buffer.ByteBuf;
+import io.reactivesocket.ConnectionSetupPayload;
+import io.reactivesocket.ReactiveSocket;
+import io.reactivesocket.ReactiveSocketConnector;
+import io.reactivex.netty.protocol.tcp.client.TcpClient;
 import org.reactivestreams.Publisher;
-import org.reactivestreams.Subscriber;
-import org.reactivestreams.Subscription;
 
 import java.net.SocketAddress;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import java.util.function.Consumer;
+import java.util.function.Function;
 
-/**
- * An implementation of {@link ReactiveSocketConnector} that creates Netty TCP ReactiveSockets.
- */
 public class TcpReactiveSocketConnector implements ReactiveSocketConnector<SocketAddress> {
-    private final ConnectionSetupPayload connectionSetupPayload;
+
+    private final ConcurrentMap<SocketAddress, TcpReactiveSocketFactory> socketFactories;
```

Comment:
This code is caching `HttpClient` instead of creating a new one for every connect. `HttpClient` has 1:1 mapping to a `SocketAddress` or a stream of `SocketAddress` but you can not connect to an arbitrary socket.

We can discuss this later but IMO we do not need `ReactiveSocketConnector` contract, we can just have `ReactiveSocketFactory`.